packages/document-grounding/src/client/api/retrieval-federated-search-across-data-repositories-api.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2024 SAP SE or an SAP affiliate company. All rights reserved.
+ *
+ * This is a generated file powered by the SAP Cloud SDK for JavaScript.
+ */
+import { OpenApiRequestBuilder } from '@sap-ai-sdk/core';
+import type { SearchInput, SearchResults } from './schema/index.js';
+/**
+ * Representation of the 'RetrievalFederatedSearchAcrossDataRepositoriesApi'.
+ * This API is part of the 'api' service.
+ */
+export const RetrievalFederatedSearchAcrossDataRepositoriesApi = {
+  /**
+   * Create a request builder for execution of post requests to the '/lm/document-grounding/retrieval/search' endpoint.
+   * @param body - Request body.
+   * @param headerParameters - Object containing the following keys: AI-Resource-Group.
+   * @returns The request builder, use the `execute()` method to trigger the request.
+   */
+  retrievalV1RetrievalEndpointsSearchDataRepositories: (
+    body: SearchInput,
```

Comment:
For reviewer, we already know that the merged spec has an issue. It mixes retrieval search body with collection/vector search. `SearchInput` should be different for retrieval and vector.

This PR is blocked due to this issue.